debvader/batch_generator.py

```diff
@@ -0,0 +1,73 @@
+from random import choice
+
+import numpy as np
+from tensorflow.keras.utils import Sequence
+
+
+class COSMOSsequence(Sequence):
+    def __init__(
+        self,
+        list_of_samples,
+        x_col_name,
+        y_col_name,
+        batch_size,
+        num_iterations_per_epoch,
+        normalization_func,
+        channel_last=False,
+    ):
+        """
+        initializes the Data generator
+
+        parameters:
+        list_of_samples: list of paths to the datafiles.
+        x_col_name: column name of data to be fed as input to the network
+        y_col_name: column name of data to be fed as target to the network
+        batch_size: sample sixe for each batch
+        num_iterations_per_epoch: number of samples (of size = batch_size) to be drawn from the sample
+        normalization_func: function to be used for normalization
+        channel_last: boolean to indicate if the the clast channel corresponds to differnet bands of the input data.
+        """
+        self.list_of_samples = list_of_samples
+        self.x_col_name = x_col_name
+        self.y_col_name = y_col_name
+        self.batch_size = batch_size
+        self.num_iterations_per_epoch = num_iterations_per_epoch
+        self.normalization_func = normalization_func
+        self.channel_last = channel_last
+
+    def __len__(self):
+        return self.num_iterations_per_epoch
+
+    def __getitem__(self, idx):  # WHY idx??
+
+        current_loop_file_name = choice(self.list_of_samples)
+        current_sample = np.load(current_loop_file_name, allow_pickle=True)
+
+        batch = np.random.choice(current_sample, size=self.batch_size, replace=False)
+        x = batch[self.x_col_name]
+        y = batch[self.y_col_name]
```

Comment:
Did you use an example to build this batch data generator ?

I am not sure I can follow the logic at the beginning of the `__getitem__`.

Reply:
From what I understood, typically a file is storing only one image and __getitem__ usually selects files(or images) whose numbers are given by the parameter `idx`.

But the way my simulations are generated, each file consists of a few hundred images. 
So I randomly select a file from which I want to load data and then extract elements from it.

Reply:
I am not sure if this is the right way, I more or less kept to the procedure with which Bastien had trained his old VAE

Reply:
So if I understand correctly, you are loading an entire file of ~100 images just to extract a couple of them, right ?

Reply:
I am extracting the entire batch of images from the file